alacritty_terminal/src/grid/mod.rs

```diff
@@ -346,8 +341,9 @@ impl<T: GridCell + PartialEq + Copy> Grid<T> {
 
                     // Remove one line from scrollback, since we just moved it to the
                     // viewport
-                    self.scroll_limit = self.scroll_limit.saturating_sub(1);
-                    self.display_offset = min(self.display_offset, self.scroll_limit);
+                    // self.decrease_scroll_limit(1);
+                    history_size -= 1;
+                    self.display_offset = min(self.display_offset, history_size);
```

Comment:
I feel like this is not the right way to do it.

Reply:
What happens if you just `decrease_scroll_limit`? Does that cause issues since `self.raw` is drained?

Reply:
Yes, you have 2 mut refs, so you're iterating over inner storage, and in the same time decrease_scroll_limit can completely change the internal structure of the internal storage, so the only way to to accumulate storage update in a loop and then update storage. However I don't know whether it's correct way to do it or not right now.